src/app/data-client.ts

```diff
@@ -53,11 +53,21 @@ export class DataClient {
         break;
       }
       dataArray.push(
-        ...dataList.map((data) => ({
-          ...data.toObject(),
-          timeRequested: data.getTimeRequested()?.toDate(),
-          timeReceived: data.getTimeReceived()?.toDate(),
-        }))
+        ...dataList.map((data) => {
+          const mdIndex = data.getMetadataIndex();
+          const mdListLength = response.getMetadataList().length;
```

Comment:
we can move this line before the loop starts, since this number will always be the same